### PR TITLE
[REEF-1849] Fix CoreCLR incompatibilities in REEF Network

### PR DESCRIPTION
**Pull request by rogancarr**
[This](url) addressed the issue by
  * Switching `System.Runtime.Caching.MemoryCache` to `Microsoft.Extensions.Caching.Memory.MemoryCache` in `NameCache`
  * Changing the `NameCacheConfiguration` to match the new configuration parameters
  * Removing unused, meaningless internal methods from `NameCache`

JIRA:
  [REEF-1849](https://issues.apache.org/jira/browse/REEF-1849)

Pull request:
  This closes #

### CODE REVIEW

**Comment by rogancarr** [comment]
Starting a discussion on the mailing list

**Comment by rogancarr** [comment]
I also created [REEF-1856](https://issues.apache.org/jira/browse/REEF-1856) to remove the `obsolete` constructor and parameters.

File: lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs [hunk 1]
```diff
@@ -38,19 +39,35 @@ internal sealed class NameCache
...
             [Parameter(typeof(NameCacheConfiguration.CacheMemoryLimit))] string memoryLimit,
             [Parameter(typeof(NameCacheConfiguration.PollingInterval))] string pollingInterval)
         {
-            var config = new NameValueCollection
+            Logger.Log(Level.Warning, "Received a parameter \"PollingInterval\" which will be interpreted as ExpirationScanFrequency.");
+            Logger.Log(Level.Warning, "Received a parameter \"CacheMemoryLimit\" which will be ignored.");
+
+            var cacheConfig = new MemoryCacheOptions
+            {
+                ExpirationScanFrequency = TimeSpan.Parse(pollingInterval)
```
Comment (dougmsft) [inline lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs]: `MemoryCacheOptions` also has Clock and `CompactOnMemoryPressure `properties. Are you assuming the default values for these fields will be reasonable?

File: lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs [hunk 2]
```diff
@@ -38,19 +39,35 @@ internal sealed class NameCache
...
+        private NameCache(
+            [Parameter(typeof(NameCacheConfiguration.CacheEntryExpiryTime))] double expirationDuration,
+            [Parameter(typeof(NameCacheConfiguration.ExpirationScanFrequency))] string expirationScanFrequency)
+        {
+            var cacheConfig = new MemoryCacheOptions
             {
-                { "pollingInterval", pollingInterval },
-                { "physicalMemoryLimitPercentage", "0" },
-                { "cacheMemoryLimitMegabytes", memoryLimit }
+                ExpirationScanFrequency = TimeSpan.Parse(expirationScanFrequency)
```
Comment (dougmsft) [inline lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs]: Same as above.

File: lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs (hunk 1 above)
Comment (rogancarr) [inline lang/cs/Org.Apache.REEF.Network/Naming/NameCache.cs]: I do assume that the defaults for these will be reasonable. I was hoping to restrict the API to as few options as possible (so as to not target this specific implementation), but given that these `MemoryCacheOptions` are for the `IMemoryCache` interface, shall I go ahead and add them in?

**Pull request closed without merging**